packages/eslint-plugin-gestalt/src/prefer-flex.js

```diff
@@ -0,0 +1,168 @@
+/**
+ * @fileoverview Encourage the use of Flex instead of Box
+ *
+ * We prefer using Flex over Box to better separate concerns and to encourage
+ * the usage of `gap`. This linter checks for usage of Box that could be Flex
+ * given the used props.
+ */
+
+// @flow strict
+import { buildProps } from './eslintASTHelpers.js';
+import { renameTagWithPropsFixer, updateGestaltImportFixer } from './eslintASTFixers.js';
+
+const sharedProps = [
+  'alignContent',
+  'alignItems',
+  'alignSelf',
+  'children',
+  'direction',
+  'flex',
+  'height',
+  'justifyContent',
+  'maxHeight',
+  'maxWidth',
+  'minHeight',
+  'minWidth',
+  'overflow',
+  'width',
+  'wrap',
+];
+
+export const errorMessage =
+  "Please use Flex for flexbox layouts. If you are wrapping children in Boxes to set margin/padding, try using Flex's `gap` prop instead!";
+
+const rule = {
+  meta: {
+    docs: {
+      description: 'Encourage usage of Flex instead of Box',
+      recommended: false,
+    },
+    fixable: 'code',
+    schema: [
+      {
+        type: 'object',
+        additionalProperties: false,
+      },
+    ],
+  },
+
+  // $FlowFixMe[unclear-type]
+  create(context: Object): Object {
```

Comment:
create(context) {

Reply:
Not sure why Flow doesn't care in the rules you've written recently, but I get `Cannot build a typed interface for this module.` 😕

Reply:
Ah nm, it was because you're declaring the ESLintRule type